dev-db/etcd/etcd-3.5.9.ebuild

```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit go-module systemd tmpfiles
+GIT_COMMIT=a603c0798
+
+DESCRIPTION="Highly-available key value store for shared configuration and service discovery"
+HOMEPAGE="https://github.com/etcd-io/etcd"
+SRC_URI="https://github.com/etcd-io/etcd/archive/v${PV}.tar.gz -> ${P}.tar.gz"
+SRC_URI+=" etcd-3.5.9-deps.tar.xz"
```

Comment:
I tried creating a dependency tarball myself for etcd-3.5.10 and found that you actually need to combine deps from multiple directories, since this project has separate modules in all of these directories found in scripts/test_lib.sh:
```bash
function module_dirs() {
  echo "api pkg raft client/pkg client/v2 client/v3 server etcdutl etcdctl tests ."
}
```

Reply:
For the 3.5.12 dependency tarball (0fca5ba6eefb64975aef368b4d6b3244b1e5066f) I've reduced the size by using `go mod vendor` for most of the dependencies (I've been doing this since https://github.com/gentoo/gentoo/pull/35773) and then just using a go-mod directory created like this:
```bash
GOMODCACHE="${PWD}/../go-mod" go mod download -modcacherw go.etcd.io/gofail@v0.1.0
```